entsoe/decorators.py

```diff
@@ -73,33 +77,66 @@ def documents_wrapper(*args, **kwargs):
                 raise NoMatchingDataError
 
             df = pd.concat(frames, sort=True)
-            df = df.loc[~df.index.duplicated(keep='first')]
+            # For same indices pick last valid value
+            if df.index.has_duplicates:
+                df = df.groupby(df.index).agg(deduplicate_documents_limited)
             return df
         return documents_wrapper
     return decorator
 
 
+def deduplicate_documents_limited(group):
+    if group.shape[0] == 1:
+        return group
+    else:
+        return group.ffill().iloc[[-1]]
+
+
 def year_limited(func):
-    """Deals with calls where you cannot query more than a year, by splitting
-    the call up in blocks per year"""
+    """Deals with calls where you cannot query more than a year,
+    by splitting the call up in blocks per year"""
 
     @wraps(func)
     def year_wrapper(*args, start=None, end=None, **kwargs):
         if start is None or end is None:
-            raise Exception('Please specify the start and end date explicity with start=<date> when calling this '
-                            'function')
-        if type(start) != pd.Timestamp or type(end) != pd.Timestamp:
-            raise Exception('Please use a timezoned pandas object for start and end')
+            raise Exception(
+                'Please specify the start and end date explicity with'
+                'start=<date> when calling this function'
+            )
+        if (
+            not isinstance(start, pd.Timestamp)
+            or not isinstance(end, pd.Timestamp)
+        ):
+            raise Exception(
+                'Please use a timezoned pandas object for start and end'
+            )
         if start.tzinfo is None or end.tzinfo is None:
-            raise Exception('Please use a timezoned pandas object for start and end')
+            raise Exception(
+                'Please use a timezoned pandas object for start and end'
+            )
 
         blocks = year_blocks(start, end)
         frames = []
         for _start, _end in blocks:
             try:
                 frame = func(*args, start=_start, end=_end, **kwargs)
+                # Due to partial matching func may return data indexed by
+                # timestamps outside _start and _end. In order to avoid
+                # (unintentionally) repeating records, frames are truncated to
+                # left-open intervals. Additionally, second disjunct forces the
+                # earliest block to be a closed interval.
+                #
+                # If there are repeating records in a single frame (e.g. due
+                # to corrections) then the result will also have them.
+                interval_mask = (
+                    ((frame.index > _start) & (frame.index <= _end))
+                    | (frame.index == start)
+                )
+                frame = frame.loc[interval_mask]
```

Comment:
A bit late to the party but I believe this change is breaking queries returning yearly data like `client.query_installed_generation_capacity(country_code, start, end, psr_type=None)`

example to recreate issue:
```python
from entsoe import EntsoePandasClient
import pandas as pd


client = EntsoePandasClient(api_key=API_KEY)

start = pd.Timestamp('20171201', tz='Europe/Brussels')
end = pd.Timestamp('20180101', tz='Europe/Brussels')
country_code = "BE"

client.query_installed_generation_capacity(
    country_code=country_code, start=start, end=end, psr_type="B01"
)
```

Expected result:
```
  Biomass
2017-01-01 00:00:00+01:00    598.0
```

Returned results:
```
Empty DataFrame
Columns: [Biomass]
Index: []
```